src/parse/value/css_function.rs

```diff
@@ -0,0 +1,395 @@
+use std::{borrow::Borrow, iter::Iterator};
+
+use codemap::Spanned;
+
+use crate::{
+    error::SassResult,
+    utils::{
+        as_hex, hex_char_for, is_name, peek_ident_no_interpolation, peek_until_closing_curly_brace,
+        peek_whitespace,
+    },
+    value::Value,
+    Token,
+};
+
+use super::super::Parser;
+
+impl<'a> Parser<'a> {
+    pub(super) fn eat_calc_args(&mut self, buf: &mut String) -> SassResult<()> {
+        buf.reserve(2);
+        buf.push('(');
+        let mut nesting = 0;
+        while let Some(tok) = self.toks.next() {
+            match tok.kind {
+                ' ' | '\t' | '\n' => {
+                    self.whitespace();
+                    buf.push(' ');
+                }
+                '#' => {
+                    if let Some(Token { kind: '{', pos }) = self.toks.peek() {
+                        self.span_before = *pos;
+                        self.toks.next();
+                        let interpolation = self.parse_interpolation()?;
+                        buf.push_str(&interpolation.node.to_css_string(interpolation.span)?);
+                    } else {
+                        buf.push('#');
+                    }
+                }
+                '(' => {
+                    nesting += 1;
+                    buf.push('(');
+                }
+                ')' => {
+                    if nesting == 0 {
+                        break;
+                    } else {
+                        nesting -= 1;
+                        buf.push(')');
+                    }
+                }
+                c => buf.push(c),
+            }
+        }
+        buf.push(')');
+        Ok(())
+    }
+
+    pub(super) fn eat_progid(&mut self) -> SassResult<String> {
+        let mut string = String::new();
+        let mut span = self.toks.peek().unwrap().pos();
+        while let Some(tok) = self.toks.next() {
+            span = span.merge(tok.pos());
+            match tok.kind {
+                'a'..='z' | 'A'..='Z' | '.' => {
+                    string.push(tok.kind);
+                }
+                '(' => {
+                    self.eat_calc_args(&mut string)?;
+                    break;
+                }
+                _ => return Err(("expected \"(\".", span).into()),
+            }
+        }
+        Ok(string)
+    }
+
+    pub(super) fn try_eat_url(&mut self) -> SassResult<Option<String>> {
+        let mut buf = String::from("url(");
+        peek_whitespace(self.toks);
+        while let Some(tok) = self.toks.peek() {
+            let kind = tok.kind;
+            self.toks.advance_cursor();
+            if kind == '!'
+                || kind == '%'
+                || kind == '&'
+                || (kind >= '*' && kind <= '~')
+                || kind as u32 >= 0x0080
+            {
```

Comment:
Why not?
```rust
            if matches!(kind, '!' | '%' | '&' | '*'..='~') || kind as u32 >= 0x0080 {
```

But I don't get why `u32`, why not `u16 >= 0x80`?

Reply:
In fact I have been planning for some time to change this `if` statement (and others like it -- see `src/parser/ident.rs`) to a `match` statement. I believe this code specifically is a port from `dart-sass`, which must use `if`.